go/private/rules/binary.bzl

```diff
@@ -574,26 +575,28 @@ exit /b %GO_EXIT_CODE%
         args.add(out)
         args.add_all(ctx.files.srcs)
 
-        # We do not use -a here as the cache drastically reduces the time spent
-        # on the second go build invocation (roughly 50% faster).
-        ctx.actions.run_shell(
-            # The value of GOCACHE/GOPATH are determined from HOME.
-            # We place them in the execroot to avoid dependency on `mktemp` and because we don't know
-            # a safe scratch space on all systems. Note that HOME must be an absolute path, otherwise the
-            # Go toolchain will write some outputs to the wrong place and the result will be uncacheable.
-            # We include an output path of this action to prevent collisions with anything else,
-            # including differently configured versions of the same target, under an unsandboxed strategy.
-            command = """
-set -eu
-export HOME="$PWD/_go_tool_binary-fake-home-${{1//\\//_}}"
-trap "{go} clean -cache" EXIT;
-{go} build -trimpath -ldflags='-buildid="" {ldflags}' -o "$1" cmd/pack
-shift
-{go} build -trimpath -ldflags='-buildid="" {ldflags}' -o "$@"
-""".format(
-                go = sdk.go.path,
-                ldflags = ctx.attr.ldflags,
-            ),
+        shell_script = ctx.actions.declare_file(name + "_build.sh")
+
+        setting = ctx.attr._use_sh_toolchain_for_bootstrap_process_wrapper[BuildSettingInfo].value
+        sh_toolchain = ctx.toolchains["@bazel_tools//tools/sh:toolchain_type"]
+        if setting and sh_toolchain:
+            ctx.actions.expand_template(
+                template = ctx.file._shell_tpl,
+                output = shell_script,
+                is_executable = True,
+                substitutions = {
+                    "#!/usr/bin/env bash": "#!{}".format(sh_toolchain.path),
+                },
+            )
+        else:
+            ctx.actions.symlink(
```

Comment:
Do we need this? It looks like we could just use the template directly in this case without having to deare and create a new file.